internal/cli/cmd/status.go

```diff
@@ -30,10 +31,18 @@ func newCmdStatus() *cobra.Command {
 
 			s, err := collector.Status(context.Background())
 			// Report the most recent status even if an error occurred.
-			fmt.Print(s.Format())
```

Comment:
As the comment indicates, we still want to print the status in case of an error. At least in case of `params.Output != "json"`.

Reply:
It's still done on line 37 (in the `else` statement).

Reply:
But the `if err != nil { fatalf() }` on [line 34](https://github.com/cilium/cilium-cli/pull/958/files#diff-20d430f1433a9b02bb41f6cd3db12b6133f75c7ba6a7ab4030d78a6bc3222fa1L34) will cause the `else` to never be reached in case of an error. I think we need to print the status here.

Reply:
Oh I see. I hadn't considered the error.

Reply:
Just pushed a change which should address it.

Reply:
I think this is still not quite right. The change added reporting of the status in the `if params.Output == "json"` branch below, however we want to report it in case of `params.Output == "summary"`. I think something like the following should work:

```go
s, err := collector.Status(context.Background())
if params.Output == "summary" {
	// Report the most recent status even if an error occurred.
	fmt.Print(s.Format())
}
if err != nil {
	fatalf("Unable to determine status:  %s", err)
}
if params.Output == "json" {
	jsonStatus, err := json.MarshalIndent(s, "", " ")
	if err != nil {
		fmt.Print(s.Format())
		fatalf("Unable to marshal status to JSON:  %s", err)
	}
	fmt.Println(string(jsonStatus))
} 
```

(and drop the `else` branch below).

Reply:
OK. That will not report status in case of error when output is set to json though.

Reply:
I believe my last push fixes this.